tests/test_component_media.py

```diff
@@ -282,6 +286,250 @@ class Media:
+        """
+        Test that media work with paths defined as instances of classes that
+        subclass 'str'.
+        """
+
+        class MyStr(str):
+            pass
+
+        class SimpleComponent(component.Component):
+            class Media:
+                css = {
+                    "all": [
+                        MyStr("path/to/style.css"),
+                        "path/to/style2.css",
+                    ],
+                    "print": [
+                        MyStr("path/to/style3.css"),
+                    ],
+                    "screen": "path/to/style4.css",
+                }
+                js = [
+                    MyStr("path/to/script.js"),
+                    "path/to/script2.js",
+                ]
+
+        comp = SimpleComponent()
+        self.assertHTMLEqual(
+            comp.render_dependencies(),
+            """
+            <link href="path/to/style.css" media="all" rel="stylesheet">
+            <link href="path/to/style2.css" media="all" rel="stylesheet">
+            <link href="path/to/style3.css" media="print" rel="stylesheet">
+            <link href="path/to/style4.css" media="screen" rel="stylesheet">
+
+            <script src="path/to/script.js"></script>
+            <script src="path/to/script2.js"></script>
+            """,
+        )
+
+    def test_path_as_object_bytes(self):
+        """
+        Test that media work with paths defined as instances of classes that
+        subclass 'bytes'.
+        """
+
+        class MyBytes(bytes):
+            pass
+
+        class SimpleComponent(component.Component):
+            class Media:
+                css = {
+                    "all": [
+                        MyBytes(b"path/to/style.css"),
+                        b"path/to/style2.css",
+                    ],
+                    "print": [
+                        MyBytes(b"path/to/style3.css"),
+                    ],
+                    "screen": b"path/to/style4.css",
+                }
+                js = [
+                    MyBytes(b"path/to/script.js"),
+                    "path/to/script2.js",
+                ]
+
+        comp = SimpleComponent()
+        self.assertHTMLEqual(
+            comp.render_dependencies(),
+            """
+            <link href="path/to/style.css" media="all" rel="stylesheet">
+            <link href="path/to/style2.css" media="all" rel="stylesheet">
+            <link href="path/to/style3.css" media="print" rel="stylesheet">
+            <link href="path/to/style4.css" media="screen" rel="stylesheet">
+
+            <script src="path/to/script.js"></script>
+            <script src="path/to/script2.js"></script>
+            """,
+        )
+
+    def test_media_custom_render_js(self):
+        class MyMedia(Media):
+            def render_js(self):
+                tags: list[str] = []
+                for path in self._js:
+                    tags.append(f'<my_script_tag src="{self.absolute_path(path)}"></my_script_tag>')
```

Comment:
`absolute_path(path)` doesn't seem to do anything?

Reply:
In that tests, `absolute_path(path)` did nothing, because neither `staticfiles` app was given, nor was `STATIC_URL` set. I've added a separate test here https://github.com/EmilStenstrom/django-components/pull/526/files#diff-7667a514b72a24c770553fe3421223dcc03252c458518310f25a5ef7746d4669R530 to the use of `static(path)` / `absolute_path(path)` when `STATIC_URL` is given.